src/util/bitvector_expr.h

```diff
@@ -517,8 +517,11 @@ inline extractbits_exprt &to_extractbits_expr(exprt &expr)
 class replication_exprt : public binary_exprt
 {
 public:
-  replication_exprt(const constant_exprt &_times, const exprt &_src)
-    : binary_exprt(_times, ID_replication, _src)
+  replication_exprt(
+    const constant_exprt &_times,
+    const exprt &_src,
+    const typet &_type)
+    : binary_exprt(_times, ID_replication, _src, _type)
```

Comment:
In keeping with many (but not all!) other expression constructors: could we instead pass the arguments by value?